pytorch_lightning/loops/optimization/optimizer_loop.py

```diff
@@ -427,7 +427,7 @@ def _training_step(self, split_batch: Any, batch_idx: int, opt_idx: int) -> Clos
             )
 
             # manually capture logged metrics
-            training_step_output = self.trainer._call_accelerator_hook("training_step", *step_kwargs.values())
```

Comment:
After the changes in this PR, there is now only one instance of `_call_accelerator_hook` left, one that calls the `on_train_end` hook on the accelerator.
cc @daniellepintz @carmocca

Reply:
do you mean `on_train_start`?
https://github.com/PyTorchLightning/pytorch-lightning/blob/3d59a2faff72aa34254b156607458f2b9acf8523/pytorch_lightning/loops/fit_loop.py#L198

Reply:
Yes

Reply:
Cool. It seems like the only reason we still need that is for the `torch.cuda.empty_cache()` line here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3d59a2faff72aa34254b156607458f2b9acf8523/pytorch_lightning/accelerators/gpu.py#L50-L53

Could we put that line into `setup` instead?

Reply:
I think we could do that yes.